corehq/apps/case_search/xpath_functions/comparison.py

```diff
@@ -23,25 +23,34 @@ def property_comparison_query(context, case_property_name_raw, op, value_raw, no
 
     case_property_name = serialize(case_property_name_raw)
     value = unwrap_value(value_raw, context)
-    # adjust the user's input date based on project timezones
-    is_user_input = False
-    try:
-        # this might be inconsistent in daylight savings situations
-        value = adjust_input_date_by_timezone(value_to_date(node, value),
-                                              get_timezone_for_domain(context.domain), op)
-        is_user_input = True
-    except (XPathFunctionException, AssertionError):
-        # AssertionError is caused by tests that use domains without a valid timezone (in get_timezeone_for_domain)
-        pass
+    if case_property_name in SPECIAL_CASE_PROPERTIES:
+        try:
+            domain = context.domain
+            if isinstance(domain, set):
+                domain = list(domain)
+            timezone = get_timezone_for_domain(domain)
+            return _create_timezone_adjusted_datetime_query(case_property_name, op, value, node, timezone)
+        except (XPathFunctionException, AssertionError):
+            # AssertionError is caused by tests that use domains without
+            # a valid timezone (in get_timezeone_for_domain)
+            pass
```

Comment:
This feels like an overbroad `except` clause (I know it was already here), especially the `AssertionError`.  As I mention above, the `XPathFunctionException` is used to tell the user they're doing something invalid (see where it's raised).  If `AssertionError` is there only to support tests, shouldn't those tests be changed?  Otherwise it seems like it might risk obscuring real issues.

Reply:
Agreed, I removed the `AssertionError` and it didn't seem to fail any tests [e2eb961](https://github.com/dimagi/commcare-hq/pull/33286/commits/e2eb9614c2d6cc0c90d0e980ba9ce145a9a901c9)

Also, I mentioned this in another response but I removed XPathFunctionException by explicitly filtering by datetime fields as you recommended https://github.com/dimagi/commcare-hq/pull/33286/commits/cff09eba50c78d5a79a5db107f5600c527db83e1

Reply:
thanks!